databox/api/src/Doctrine/Delete/CollectionDelete.php

```diff
@@ -75,6 +76,23 @@ private function doDelete(string $collectionId): void
             $this->em->clear();
         }
 
+        /** @var Collection $collection */
+        $collection = $this->em->find(Collection::class, $collectionId);
+        if (!$collection instanceof Collection) {
+            throw new \InvalidArgumentException(sprintf('Collection "%s" not found for deletion', $collectionId));
+        }
+
+        if($collection->getStoryAsset() !== null) {
+            /** @var AssetRepository $assetRepository */
+            $assetRepository = $this->em->getRepository(Asset::class);
+            foreach($assetRepository->findByStoryCollectionIds([$collectionId]) as $storyAsset) {
```

Comment:
How can we have many storyAssets? Accessing `$collection->getStoryAsset()` should be sufficient?